zabbix_metrics_watcher.py

```diff
@@ -214,14 +217,32 @@ def run(zabbix_host, osbs_master, config, instance):
 
             logger.info("Found build %s in %s", build_name, status)
             build = Build(build_name, cmd_base)
-            if status == 'Pending':
+            if status == 'New':
+                now = datetime.datetime.now()
+                if build_name not in builds_in_new:
+                    builds_in_new[build_name] = now
+                spent_in_new = [(now - start).total_seconds() for start in builds_in_new.values()]
+                max_spent_in_new = max(spent_in_new)
+                if spent_in_new and max_spent_in_new > STUCK_IN_NEW_THRESHOLD_SECS:
+                    logger.info("%s build(s) are in New for, longest: %s sec",
+                                len(builds_in_new), max_spent_in_new)
+                    _send_zabbix_message(zabbix_host, osbs_master, "new_duration", max_spent_in_new)
+
+            elif status == 'Pending':
+                builds_in_new = builds_in_new.pop(build_name, {})
+                if not builds_in_new:
+                    _send_zabbix_message(zabbix_host, osbs_master, "new_duration", 0)
```

Comment:
You have `builds_in_new = builds_in_new.pop(build_name, {})` which looks like it will re-assign builds_in_new, possibly to an integer instead of a dict.

Let's be really clear about what `new_duration` is supposed to represent though. Zabbix items are just data points, 'fine or not fine' is what alerts are for.